server/sso/index.js

```diff
@@ -3,124 +3,137 @@ const base64url = require('base64url');
+
+        const cookies = resp.headers['set-cookie'];
+        let parsedCookie;
+
+        for (let i = cookies.length - 1; i >= 0; i -= 1) {
+          parsedCookie = cookie.parse(cookies[i]);
+
+          if (parsedCookie['CA.OVH.SES']) {
+            res.cookie('CA.OVH.SES', parsedCookie['CA.OVH.SES'], { path: '/', httpOnly: true });
+          }
+          if (parsedCookie.SESSION) {
+            res.cookie('SESSION', parsedCookie.SESSION, { path: '/', httpOnly: true });
+          }
+          if (parsedCookie.USERID) {
+            res.cookie('USERID', parsedCookie.USERID, { path: '/' });
+          }
         }
-      }
 
-      console.log('[SSO] - Logged');
+        console.log('[SSO] - Logged');
 
-      return resolve(resp.headers.location);
+        return resolve(resp.headers.location);
+      });
     });
-  });
 
-  return res.redirect(location);
-}
+    return res.redirect(location);
+  }
 
-async function auth(req, res) {
-  const origin = req.headers.host;
-  const protocol = req.protocol || 'http';
-  const headers = {
-    contentType: 'application/json',
-  };
-  headers.host = config.ssoAuth.host;
-
-  const redirectionUrl = await new Promise((resolve) => {
-    proxy.post({
-      url: config.ssoAuth.devLoginUrl,
-      proxy: config.proxy ? config.proxy.host : null,
-      headers,
-      followRedirect: false,
-      gzip: true,
-      json: {
-        callbackUrl: `${protocol}://${origin}/auth/check`,
-      },
-    }, (err, resp, data) => {
-      if (err) {
-        return resp.status(500);
-      }
+  async auth(req, res) {
+    const origin = req.headers.host;
+    const protocol = req.protocol || 'http';
+    const headers = {
+      contentType: 'application/json',
+    };
+    headers.host = this.config.host;
+
+    const redirectionUrl = await new Promise((resolve) => {
+      proxy.post({
+        url: this.config.devLoginUrl,
+        proxy: this.config.proxy ? this.config.proxy.host : null,
+        headers,
+        followRedirect: false,
+        gzip: true,
+        json: {
+          callbackUrl: `${protocol}://${origin}/auth/check`,
+        },
+      }, (err, resp, data) => {
+        if (err) {
+          return resp.status(500);
+        }
 
-      return resolve(data.data.url);
+        return resolve(data.data.url);
+      });
     });
-  });
 
-  res.redirect(redirectionUrl);
-}
+    res.redirect(redirectionUrl);
+  }
 
-function checkAuth(req, res) {
-  const { headers } = req;
-  headers.host = config.ssoAuth.host;
+  checkAuth(req, res) {
+    const { headers } = req;
+    headers.host = this.config.host;
 
-  let cookies = [];
+    let cookies = [];
 
-  try {
-    cookies = JSON.parse(base64url.decode(req.query.data));
+    try {
+      cookies = JSON.parse(base64url.decode(req.query.data));
 
-    if (Array.isArray(cookies.cookies)) {
-      cookies.cookies.forEach((c) => {
-        const parsedCookie = cookie.parse(c);
+      if (Array.isArray(cookies.cookies)) {
```

Comment:
The cookies have cookies themselves ? I guess this is something we don't have control on ?